tox.ini

```diff
@@ -18,7 +18,7 @@ whitelist_externals =
 commands =
     pytest tests {posargs} \
         -m "not slow" \
-        --cov-fail-under=70 \
+        --cov-fail-under=92.4 \
```

Comment:
why not 90 like in the `codecov.yml`

Reply:
It's a bit of a dirty hack, to be honest. What I want to achieve is that test coverage **does not drop**. But locally, you cannot easily compare test coverage with that of a previous commit on the master branch. So I picked a value super close to the current test coverage.

Note that he test coverage of `tox -e py` is slightly lower than Codecov, because not all tests are run.

Reply:
but then you would have to update that value all the time. Can't we just let codecov handle that?

Reply:
Well, you don't _have_ to. The way I see it is: codecov already handles that, yes, but you can use `tox` to quickly check if the value has dropped. The only potential problem is that `tox` fails when you run things locally and you implemented new functionality, but this simply means you should write more tests.

For the record, I extract these commits from ongoing work where I wanted to do precisely this.

Reply:
You have to adjust if you want the rate to never drop. Personally I think we should also not cling to the principle that the test rate should never drop. The most important part about tests is that they actually test the critical behavior. I could write a lot of bogus tests that increase the coverage but fail to catch problems. Hence I would not take this coverage value too serious, being 90+ is fine in my current view.

Reply:
Okay, let's stick with 90 then. Note that this PR was originally motivated by #138